File: algorithms/determinant.py
from math import * 

def getSubMatrix( matrix, k, m ):

    n = len(matrix)
    subMatrix = []
    arr = []
    for i in range(n):
        for j in range(n):
            if (i == k or j == m) :
                continue
            else :
                arr.append(matrix[i][j])
        if len(arr) == 0 : 
            continue
        else :
            subMatrix.append(arr)
            arr = []
    return subMatrix

def det(matrix):
    sum = i = 0
    try :
        n = len(matrix)
    except TypeError:
        return matrix
    if (len(matrix[0]) == 1) :
        return matrix[0][0]
    for j in range(n):
        sum += pow(-1, i+j)*matrix[i][j]*det(getSubMatrix(matrix, i, j))
    return sum

print ( det([[1, 1], [1, 2]]) )

File: algorithms/determinant.js

function getSubMatrix(matrix, k, m){
    const n = matrix.length;
    const subMatrix = new Array();
    let arr = new Array();
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (i === k || j === m) {
                continue;
            } else {
                arr.push(matrix[i][j]);
            } 
        }
        if (arr.length === 0) continue;
        else {
            subMatrix.push(arr);
            arr = new Array();
        }
    }
    return subMatrix;
}

function det(matrix) {
    let sum = 0, i = 0;
    const n = matrix.length;
    if( !isNaN(matrix) ) return matrix;
    if( matrix[0].length === 1 ) return matrix[0][0] ; 
    for (let j = 0; j < n; j++) {
        sum += Math.pow(-1, i+j)*matrix[i][j]*det(getSubMatrix(matrix, i, j));
    } 
    return sum;
}

console.log(det([[1, 1], [1, 2]]));
